optimism-haproxy.cfg

```diff
@@ -0,0 +1,118 @@
+#---------------------------------------------------------------------
+# Global settings
+#---------------------------------------------------------------------
+global
+    log                 stdout format raw local0
+    maxconn             20000
+    user                haproxy
+    group               haproxy
+    ssl-default-server-options force-tlsv12
+    ssl-default-bind-options force-tlsv12
+    ca-base             /etc/ssl/certs
+    external-check
+    insecure-fork-wanted
+
+#---------------------------------------------------------------------
+# common defaults that all the 'listen' and 'backend' sections will
+# use if not designated in their block
+#---------------------------------------------------------------------
+defaults
+    mode                 http
+    option               httplog
+    option               httpclose
+    log                  global
+    option               dontlognull
+    timeout client       5m
+
+#---------------------------------------------------------------------
+# dedicated stats page
+#---------------------------------------------------------------------
+listen stats
+    mode http
+    bind :22222
+    stats enable
+    stats uri            /haproxy?stats
+    stats realm          Haproxy\ Statistics
+    stats auth           admin:MYPW
+    stats refresh        30s
+
+#---------------------------------------------------------------------
+# main frontend which proxys to the backends
+#---------------------------------------------------------------------
+frontend main_http_listen
+     bind                *:80
+     http-request wait-for-body time 1s if METH_POST
+
+#---------------------------------------------------------------------
+# Common HAProxy nodes configuration
+#---------------------------------------------------------------------
+
+# -------------------------------
+# ACLs
+# -------------------------------
+
+acl acl_tx           req.body -i -m sub eth_sendRawTransaction 
```

Comment:
Could you explain what this line is doing? Is this a string check on the request body? Haproxy can't parse the json into key/value, can it?

We devised [proxyd](https://github.com/ethereum-optimism/optimism/tree/develop/go/proxyd) to handle these situations 😅

Reply:
Yes it's a string check. If it's a tx, send to official sequencer; everything else goes to the replica.

This works fine for RPC and not at all for WS, because WS requests happen after connect and upgrade. 

Can `proxyd` handle WS routing? That'd be quite amazing, we (Chainlink NOs) really need that and don't have it.

Reply:
> Can proxyd handle WS routing? 
That's part of the feature list, but we're not using websockets enough to make it a priority atm.